src/components/Modals/ModalPost/ModalPost.tsx

```diff
@@ -1,72 +1,141 @@
 import './ModalPost.css';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import { useDispatch, useSelector } from 'react-redux';
-import { getUser, getModal } from '../../../redux/selectors';
+import { getUser, getModal, getPosts } from '../../../redux/selectors';
 import { Modal } from '../../Modal/Modal';
-import { ModalPostProps } from '../../../types/post';
+import { newPostData, updatedPostData } from '../../../types/post';
 import Author from '../../Author/Author';
 import Likes from '../../Likes/Likes';
-import { addPost, updatePost } from '../../../redux/postSlice';
-
-// TODO: correct date format in post__date
-
-function ModalPost({
-  id,
-  text,
-  photoUrl,
-  authors,
-  likes,
-  createdAt,
-  onClose,
-}: ModalPostProps) {
+import { addPost, editPost, removePost } from '../../../redux/postSlice';
+import { formatTime } from '../../../utils/helpers';
+import {
+  createPost,
+  deletePosts,
+  getUserInfo,
+  updatePost,
+} from '../../../utils/api';
+import { setUser } from '../../../redux/userSlice';
+
+export type ModalPostProps = {
+  postId: string;
+  onClose: () => void;
+  newPost?: {
+    text: string;
+    photoUrl: string;
+    authors: {
+      host: {
+        _id: string;
+        name: string;
+        avatar: string;
+      };
+      resident: {
+        _id: string;
+        name: string;
+        avatar: string;
+        species: string;
+      };
+    };
+    createdAt: string;
+    likes: [];
+  };
+};
+
+function ModalPost({ postId, onClose, newPost }: ModalPostProps) {
   const navigate = useNavigate();
   const dispatch = useDispatch();
-  const currentUser = useSelector(getUser);
-  const modalIsActive = useSelector(getModal);
+  const user = useSelector(getUser);
+  const posts = useSelector(getPosts);
+  const modal = useSelector(getModal);
+  const [postText, setPostText] = useState('');
   const [postModalMode, setPostModalMode] = useState(
-    modalIsActive === 'add-post-next' ? 'edit' : 'view'
+    modal.modalIsActive === 'add-post-next' ? 'edit' : 'view'
   );
-  const [postText, setPostText] = useState(text);
+
+  const post = newPost || posts.find((post) => post._id === postId);
+
+  useEffect(() => {
+    if (post) {
+      setPostText(post.text);
+    }
+  }, [post]);
+
+  if (!post) {
+    return <p>Post not found</p>;
+  }
+
+  const { photoUrl, authors, likes, createdAt } = post;
+
+  const postingTime = formatTime(new Date(createdAt));
+  const isToday =
+    createdAt.slice(0, 10) === new Date().toISOString().slice(0, 10);
 
   const handleEditClick = () => {
-    console.log('edit');
     setPostModalMode('edit');
   };
 
   const handleTextChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
     setPostText(e.target.value);
   };
 
+  const handleDeleteClick = () => {
+    deletePosts(localStorage.jwt, postId);
+    dispatch(removePost({ _id: postId }));
+    onClose();
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**